resolwe_bio/processes/samtools/samtools_view.py

```diff
@@ -0,0 +1,227 @@
+                hidden="!advanced.subsample",
+            )
+            threads = IntegerField(
+                label="Number of threads",
+                default=2,
+                description="Number of BAM compression threads to use in addition "
+                "to main thread.",
+            )
+
+        advanced = GroupField(AdvancedOptions, label="Advanced options")
+
+    class Output:
+        """Output fields for SamtoolsView."""
+
+        bam = FileField(label="Output BAM file")
+        bai = FileField(label="Ouput index file")
+        stats = FileField(label="Alignment statistics", required=False)
+        build = StringField(label="Build")
+        species = StringField(label="Species")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+
+        name = Path(inputs.bam.output.bam.path).stem + "_subset"
+        bam_name = f"{name}.bam"
+        bai_name = f"{bam_name}.bai"
+        stats = f"{name}_stats.txt"
+
+        if not inputs.region and not inputs.bedfile and not inputs.advanced.only_header:
+            self.error("No region or BED file specified.")
+
+        if inputs.bedfile:
+            if inputs.bedfile.output.species != inputs.bam.output.species:
+                self.error(
+                    "Input BAM file and BED file are of different species. "
+                    f"BAM file is from {inputs.bam.output.species}, "
+                    f"while BED file is from {inputs.bedfile.output.species}."
+                )
+            if inputs.bedfile.output.build != inputs.bam.output.build:
+                self.error(
+                    "Input BAM file and BED file have different genome build, "
+                    "but it should be the same. BAM file has build "
+                    f"{inputs.bam.output.build}, while BED file has build "
+                    f"{inputs.bedfile.output.build}."
+                )
+
+        if inputs.advanced.only_header:
+
+            return_code, stdout, stderr = Cmd["samtools"]["view"][
+                "-Hbo", bam_name, inputs.bam.output.bam.path
+            ] & TEE(retcode=None)
+            if return_code:
+                self.error(f"Samtools view failed. {stdout}, {stderr}")
+
+            return_code, stdout, stderr = Cmd["samtools"]["index"][bam_name] & TEE(
+                retcode=None
+            )
+            if return_code:
+                self.error(f"Samtools index failed. {stdout}, {stderr}")
+
+            self.warning("Only header is present in the output BAM file.")
+
+        else:
+
+            input_options = [
+                "-b",
+                "--threads",
+                max(inputs.advanced.threads, self.requirements.resources.cores),
+            ]
+
+            if inputs.advanced.include_header:
+                input_options.append("-h")
+
+            if inputs.advanced.subsample:
+                if not 0 <= inputs.advanced.subsample <= 1:
```

Comment:
This should not a necessary check if the range limit on the input field is used. Please double check, as the range limit might work only on the front-end.